### PR TITLE
nixos/pam: Allow password changing via sssd

### PR DESCRIPTION
**Pull request by s1341**
###### Description of changes

Modify the pam configuration for sssd to not use `use_authtok`, as the it swallows password change prompts, and otherwise doesn't appear to make a difference.

Also expanded the nixos `sssd-ldap` tests to test login and password change scenarios. This required adding TLS certificates to the openldap server in `sssd-ldap`, as you cannot login without TLS.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by s1341** [comment]
@ofborg eval

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Testing the change, some minor formatting issues

**Comment by RaitoBezarius** [comment]
Add `sssd-ldap` as a test for PAM please so that it gets built automatically too

File: nixos/tests/sssd-ldap.nix [hunk 1]
```diff
@@ -6,96 +6,165 @@ let
...
+  testNewPassword = "barfoo";
+in
+  import ./make-test-python.nix ({pkgs, ...}: {
+    name = "sssd-ldap";
 
-  meta = with pkgs.lib.maintainers; {
-    maintainers = [ bbigras ];
-  };
+    meta = with pkgs.lib.maintainers; {
+      maintainers = [bbigras];
```
[the remark this answers is not in the document]
Comment (s1341) [reply nixos/tests/sssd-ldap.nix:16]: woops. Sorry about that. Should I add myself too?

[the remark this answers is not in the document]
Comment (RaitoBezarius) [reply nixos/tests/sssd-ldap.nix]: please that would be awesome

**Comment by s1341** [comment]
Thanks!

**Pull request merged**